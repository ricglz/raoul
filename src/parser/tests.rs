use super::*;
use std::fs::read_dir;

#[test]
fn valid_files() {
    let paths = read_dir("examples/valid").unwrap();
    for path in paths {
        let file_path = path.expect("File must exist").path();
        let file = file_path.to_str().unwrap();
        if file == "examples/valid/complete.ra" {
            continue;
        }
        let program = std::fs::read_to_string(file).expect(file);
        println!("Testing {:?}", file);
        let res = parse(&program, true);
        assert!(res.is_ok());
        insta::assert_debug_snapshot!(res.unwrap());
    }
}

#[test]
fn invalid_file() {
    let filename = "examples/invalid/syntax/syntax-error.ra";
    let program = std::fs::read_to_string(filename).expect(filename);
    let res = parse(&program, true);
    assert!(res.is_err());
    insta::assert_display_snapshot!(res.unwrap_err());
}
