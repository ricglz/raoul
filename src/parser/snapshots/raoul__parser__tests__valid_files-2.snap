---
source: src/parser/tests.rs
expression: res.unwrap()
---
Main(([
    Function(helper, VOID, [Argument(INT, a), Argument(FLOAT, b), Argument(STRING, c), Argument(BOOL, d)], []),
], [
    Assignment(false, Id(a), Integer(1)),
    Assignment(false, Id(b), Float(1)),
    Assignment(false, Id(c), String(my string)),
    Assignment(false, Id(d), Bool(false)),
    Write([Id(a), Id(b), Id(c), Id(d)]),
]))
